src/azure-cli-core/azure/cli/core/context.py

```diff
@@ -0,0 +1,158 @@
+#---------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+#---------------------------------------------------------------------------------------------
+
+import os
```

Comment:
It needs to be part of CLI core as almost all operations require knowledge of 'the current cloud' and this is saved in context.

The user will not be exposed to context though unless they need it for their particular scenario.